stl/inc/bit

```diff
@@ -125,10 +234,16 @@ _NODISCARD constexpr int countr_one(const _Ty _Val) noexcept {
 
 template <class _Ty, enable_if_t<_Is_standard_unsigned_integer<_Ty>, int> _Enabled = 0>
 _NODISCARD constexpr int popcount(const _Ty _Val) noexcept {
-    if constexpr (sizeof(_Ty) <= sizeof(unsigned int)) {
-        return __builtin_popcount(_Val);
+    if (is_constant_evaluated()) {
+        return _Popcount_helper(_Val);
     } else {
-        return __builtin_popcountll(_Val);
+#if defined(_M_IX86) || defined(_M_X64)
+        return _Checked_x86_x86_64_popcount(_Val);
+#else
+        // arm NEON has a popcount instruction, but we're not going to use that
+        // yet.

```

Comment:
Should this be a `TRANSITION` comment or something? Why aren't we using it yet?
```suggestion
        // arm NEON has a popcount instruction, but we're not going to use that yet.
```

Reply:
we're not using it because each intrinsic we use takes a bunch of time to research how/when/on what chips it works, and it would require more faffing about with runtime cpu feature detection. I don't think it's so much a TRANSITION as a, this would be kinda cool to have, maybe it should be though